### PR TITLE
Added java.time.Period assertions

### LINKED ISSUE #1853: Add assertThat for java.time.Period
#### Summary

Similar to #1650 it would be great to support `java.time.Period`

#### Example

```java
Period period = Period.of(1,1,1)
assertThat(period)
    .hasYear(1)
    .hasMonth(1)
    .hasDays(1);
```

If you are agree with this I can work on it.

### PR DESCRIPTION
**Pull request by HaydenMeloche**
Added assertions:
* hasDays()
* hasMonths()
* hasYears()
* isPositive()
* isNegative()

#### Check List:
* Fixes #1853 
* Unit tests : yes
* Javadoc with a code example (on API only) : yes

### CODE REVIEW

File: src/test/java/org/assertj/core/api/period/PeriodAssert_hasMonths_Test.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.error.ShouldHavePeriod.shouldHaveMonths;
+import static org.assertj.core.util.AssertionsUtil.assertThatAssertionErrorIsThrownBy;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+/**
+ * @author Hayden Meloche
+ */
+@DisplayName("PeriodAssert hasMonths")
+class PeriodAssert_hasMonths_Test {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_hasMonths_Test.java:30]: same comments as in `PeriodAssert_hasDays_Test`

File: src/test/java/org/assertj/core/api/period/PeriodAssert_hasMonths_Test.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+  @Test
+  void should_pass_if_period_has_matching_Months() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_hasMonths_Test.java:33]: matching_Months -> expected_months

File: src/test/java/org/assertj/core/api/period/PeriodAssert_hasDays_Test.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+/**
+ * @author Hayden Meloche
+ */
+@DisplayName("PeriodAssert hasDays")
+class PeriodAssert_hasDays_Test {
+
+  @Test
+  void should_pass_if_period_has_matching_days() {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_hasDays_Test.java:33]: replace `matching` by `expected`

File: src/test/java/org/assertj/core/api/period/PeriodAssert_hasYears_Test.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.error.ShouldHavePeriod.shouldHaveYears;
+import static org.assertj.core.util.AssertionsUtil.assertThatAssertionErrorIsThrownBy;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+/**
+ * @author Hayden Meloche
+ */
+@DisplayName("PeriodAssert hasYears")
+class PeriodAssert_hasYears_Test {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_hasYears_Test.java:30]: same comments as in `PeriodAssert_hasDays_Test`

File: src/test/java/org/assertj/core/api/period/PeriodAssert_isNegative_Test.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.error.ShouldBePeriod.shouldBeNegative;
+import static org.assertj.core.util.AssertionsUtil.assertThatAssertionErrorIsThrownBy;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+/**
+ * @author Hayden Meloche
+ */
+@DisplayName("PeriodAssert isNegative")
+class PeriodAssert_isNegative_Test {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_isNegative_Test.java:30]: same comments as in `PeriodAssert_hasDays_Test`

File: src/test/java/org/assertj/core/api/period/PeriodAssert_isPositive_Test.java [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    // WHEN
+    ThrowableAssert.ThrowingCallable code = () -> assertThat(period).isPositive();
+    // THEN
+    assertThatAssertionErrorIsThrownBy(code).withMessage(shouldBePositive(period).create());
+  }
+
+  @Test
+  void should_fail_if_period_is_zero() {
+    // GIVEN
+    Period period = Period.ofMonths(0);
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_isPositive_Test.java:64]: use Period.ZERO

File: src/test/java/org/assertj/core/api/period/PeriodAssert_isPositive_Test.java [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+import static org.assertj.core.error.ShouldBePeriod.shouldBePositive;
+import static org.assertj.core.error.ShouldHavePeriod.shouldHaveMonths;
+import static org.assertj.core.util.AssertionsUtil.assertThatAssertionErrorIsThrownBy;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+
+/**
+ * @author Hayden Meloche
+ */
+@DisplayName("PeriodAssert isPositive")
+class PeriodAssert_isPositive_Test {
```
Comment (joel-costigliola) [inline src/test/java/org/assertj/core/api/period/PeriodAssert_isPositive_Test.java:31]: same comments as in `PeriodAssert_hasDays_Test`

**Comment by HaydenMeloche** [comment]
Thanks @joel-costigliola 😄

**Comment by joel-costigliola** [comment]
@HaydenMeloche thanks, I realize there are missing tests for `ShouldHavePeriod` and `ShouldBePeriod`, use `ShouldHaveDuration_create_test` as a reference for implementing them

**Comment by HaydenMeloche** [comment]
@joel-costigliola Sorry about that, I added the missing the tests.

**Comment by joel-costigliola** [comment]
Integrated thanks @HaydenMeloche!

**Pull request closed without merging**